pkg/tsdb/zipkin/zipkin.go

```diff
@@ -0,0 +1,81 @@
+package zipkin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+
+	"github.com/grafana/grafana/pkg/infra/httpclient"
+)
+
+type Service struct {
+	im instancemgmt.InstanceManager
+}
+
+func ProvideService(httpClientProvider httpclient.Provider) *Service {
+	return &Service{
+		im: datasource.NewInstanceManager(newInstanceSettings(httpClientProvider)),
+	}
+}
+
+type datasourceInfo struct {
+	ZipkinClient ZipkinClient
+}
+
+func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.InstanceFactoryFunc {
+	return func(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		httpClientOptions, err := settings.HTTPClientOptions(ctx)
+		if err != nil {
+			return nil, backend.DownstreamError(fmt.Errorf("error reading settings: %w", err))
+		}
+
+		httpClient, err := httpClientProvider.New(httpClientOptions)
+		if err != nil {
+			return nil, fmt.Errorf("error creating http client: %w", err)
+		}
+
+		if settings.URL == "" {
+			return nil, backend.DownstreamError(fmt.Errorf("error reading settings: url is empty"))
+		}
+
+		logger := backend.NewLoggerWith("logger", "tsdb.zipkin")
```

Comment:
I think you may be better off here to make this global (still unexported) and then use `logger.FromContext(ctx)` here 🤔

Reply:
Fixed in https://github.com/grafana/grafana/pull/96031/commits/25a75d5e9ca5fe6976636f6ac3af16ed9dc83571